src/interface.jl

```diff
@@ -202,16 +202,16 @@ end
 
 WARNING: experimental functionality. Interface subject to change without warning!
 
-Like other methods of `value_and_pullback!!`, but makes use of the `cache` object in order
-to avoid having to re-allocate various tangent objects repeatedly.
+Like other methods of `value_and_pullback!!`, but makes use of the `cache` object returned
+by [`prepare_pullback_cache`](@ref) in order to avoid having to re-allocate various tangent
+objects repeatedly. You must ensure that `f` and `x` are the same types and sizes as those
+used to construct `cache`.
 
-You must ensure that `f` and `x` are the same types and sizes as those used to construct
-`cache`.
-
-Warning: any mutable components of values returned by `value_and_gradient!!` will be mutated
-if you run this function again with different arguments. Therefore, if you need to keep the
-values returned by this function around over multiple calls to this function with the same
-`cache`, you should take a copy of them before calling again.
+Warning: `cache` owns any mutable state returned by this function, meaning that mutable
+components of values returned by it will be mutated if you run this function again with
+different arguments. Therefore, if you need to keep the values returned by this function
+around over multiple calls to this function with the same `cache`, you should take a copy
```

Comment:
Maybe specify "or `deepcopy`"? I think this may come back to bite us more than once.
Otherwise good to go, thanks!

Reply:
Good point.